detox/test/e2e/16.location.test.js

```diff
@@ -22,17 +30,29 @@ describe(':ios: location', () => {
     await expect(element(by.id('error'))).toBeVisible();
   });
 
-  it('Should receive location (20,20)', async () => {
+  it(':ios: Should receive location (20,20)', async () => {
     if (!await isFbsimctlInstalled()) {
       return;
     }
     await device.relaunchApp({ permissions: { location: 'always' } });
-    await device.setLocation(20.1, 20.2);
+    await device.setLocation(lat, long);
+
+    await element(by.text('Location')).tap();
+    await element(by.id('getLocationButton')).tap();
+    await waitFor(element(by.text(`Latitude: ${lat}`))).toBeVisible().withTimeout(3000);
```

Comment:
Why so we need manual sync here? Is something not syncing correctly? 
As a rule of thumb, we should never use waitFor, especially in Detox's own test suite

Reply:
Can remove, however it's an asynchronous process relying on location hardware/emulation so there might be some random time. This is essentially copied from the test that was originally there for iOS.

Reply:
> This is essentially copied from the test that was originally there for iOS.

@rotemmiz, this.

I was under impression that there were no idling resources for geo implemented yet.

Reply:
I see now (was going over this on my phone yesterday, couldn't get the full context). If it's the same test, don't copy code, merge the two together.
Something like
```js
  it('Should receive location (20,20)', async () => {
    if (device.platform() === 'ios' && !await isFbsimctlInstalled()) {
      return;
    }
    await device.relaunchApp({ permissions: { location: 'always' } });
    await device.setLocation(lat, long);

    await element(by.text('Location')).tap();
    await element(by.id('getLocationButton')).tap();
    await waitFor(element(by.text(`Latitude: ${lat}`))).toBeVisible().withTimeout(3000);

    await expect(element(by.text(`Latitude: ${lat}`))).toBeVisible();
    await expect(element(by.text(`Longitude: ${long}`))).toBeVisible();
  });
```

Reply:
Legit! Thx

Reply:
@shockdesign , please change

Reply:
@noomorph have updated